tests/dist/chaining/test_openmp.cpp

```diff
@@ -0,0 +1,45 @@
+#include <catch2/catch.hpp>
```

Comment:
Slightly unrelated point: why do this tests live in `tests/dist/chaining`? The name `chaining` seems a bit odd to me.